src/PathFinderUtils.js

```diff
@@ -184,6 +202,48 @@ function findPath(params, onUpdate, onDone, onError) {
   }
 
   return terminate;
-}
+};
 
-export default findPath;
+export var findAddress = async (path, segwit, coin, onError) => {
+  try {
+    var comm = await Dongle.init();
+    await Dongle.setCoinVersion(comm, Networks[coin]);
+    const xpub58 = await initialize(
+      path.split("/")[1].replace("'", ""),
+      path.split("/")[2].replace("'", ""),
+      segwit
+    );
+    var script = segwit
+      ? Networks[coin].bitcoinjs.scriptHash
+      : Networks[coin].bitcoinjs.pubKeyHash;
+    var hdnode = bitcoin.HDNode.fromBase58(xpub58, Networks[coin].bitcoinjs);
+    var pubKeyToSegwitAddress = (pubKey, scriptVersion, segwit) => {
+      var script = [0x00, 0x14].concat(
+        Array.from(bitcoin.crypto.hash160(pubKey))
+      );
+      var hash160 = bitcoin.crypto.hash160(script);
+      return bitcoin.address.toBase58Check(hash160, scriptVersion);
+    };
+
+    var getPublicAddress = (hdnode, path, script, segwit) => {
+      hdnode = hdnode.derivePath(
+        path
+          .split("/")
+          .splice(3, 2)
+          .join("/")
+      );
+      if (!segwit) {
+        return hdnode.getAddress().toString();
+      } else {
+        return pubKeyToSegwitAddress(
+          hdnode.getPublicKeyBuffer(),
+          script,
+          segwit
+        );
+      }
+    };
+    return await getPublicAddress(hdnode, path, script, segwit);
+  } catch (e) {
+    onError(e);
```

Comment:
you probably shouldn't `try{catch` in this function but just the error flow in the Promise!. that removes the need to pass `onError` to avoid mixing 2 paradigm together.
like on the other side, it's weird to do

```js
findAddress(p,s,c,error => {...}).then(result => {...})
```

where you could just


```js
findAddress(p,s,c).then(result => {...}, error => {...})
```


basically, let the error flow until you want to handle it, typically at the UI level.